src/torchcodec/_core/Encoder.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+#include <torch/types.h>
+#include "src/torchcodec/_core/FFMPEGCommon.h"
+
+namespace facebook::torchcodec {
+class Encoder {
+ public:
+  ~Encoder();
+
+  Encoder(const torch::Tensor wf, int sampleRate, std::string_view fileName);
+  void encode();
+
+ private:
+  void encode_inner_loop(
+      AutoAVPacket& autoAVPacket,
+      const UniqueAVFrame& avFrame);
+
+  UniqueAVFormatContextForEncoding avFormatContext_;
+  UniqueAVCodecContext avCodecContext_;
+  AVStream* avStream_;
```

Comment:
We store the stream pointer, but we don't actually use the stream itself anymore, we only reference its index and timebase. It may be clearer what's going on if we just store those.

Reply:
I like to have the avStream_ field because it makes it obvious where `timebase` and `index` come from, without having to bubble up all the way to the constructor code. I'll switch it for `timeBase_` and `streamIndex_` fields for now, but we might revisit if we end up having to rely on more attributes from the AVStream?